### ISSUE TITLE
About generating multiple inpainting results.

### ISSUE THREAD

**Issue by duzhenjiang113** [raywzy/ICT#15]
Hello, wonderful work! I have a question. How does your method ensure that the results of completion are different?

**Comment by raywzy** [comment]
Hi,
The diversity comes from sampling, whose category distribution is the prediction of transformer.

**Comment by duzhenjiang113** [comment]
def sample_mask_all(model, context, length, num_sample=1, temperature=1.0, top_k=None, mask=None,no_bar=False):

    # output = torch.zeros(num_sample,length,dtype=torch.long).cuda()
    output = context.cuda()
    mask = mask.cuda()

    model.eval()
    with torch.no_grad():

        logits,_ = model(output,masks=mask)

        if no_bar:
            looper=range(length)
        else:
            looper=tqdm(range(length), leave=False)
        for i in looper:

            if mask[0,i] == 0:
                continue
            logits_i = logits[:, i, :] / temperature
            if top_k is not None:
                logits_i = top_k_logits(logits_i, top_k)
            probs = F.softmax(logits_i, dim=-1)
            pred = torch.multinomial(probs, num_samples=1)
            output[:,i] = pred[:,0]

    return output

From this point of view, it seems that the results of all the completions are directly generated. Can this generate multiple different completion results for the same input image?

**Comment by raywzy** [comment]
Hi,
Sure. The sample number depends on the batch size (first dimension). Even though the `num_samples` is set to 1, this code will randomly sample different values for each batch.

**Comment by duzhenjiang113** [comment]
> Hi, Sure. The sample number depends on the batch size (first dimension). Even though the `num_samples` is set to 1, this code will randomly sample different values for each batch.

okay, thanks
